apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/VolatileResourceManager.cs

```diff
@@ -115,8 +115,15 @@ public void SinglePhaseCommit(SinglePhaseEnlistment singlePhaseEnlistment)
         {
             try
             {
-                ExecuteHelper.WithErrorTranslationAndProfiling(() =>
-                    _transaction?.Commit(), "VolatileResourceManager.Commit");
+                // If we have no changes, then just tell the outer transaction that everything is good.
+                // This can happen with a readonly transaction or a write transaction where we never
+                // executed any mutations.
```

Comment:
Readonly transactions do not take locks -- but readwrite transactions may actually have locks if a read was previously done on it.  So while we do a prewarm that would create a new transaction, thats a bit fragile to rely on.  So I'm going to narrow the logic here to only ignore readonly transactions -- so empty readwrite tx will still get committed.